jina/optimizers/__init__.py

```diff
@@ -169,4 +175,19 @@ def optimize_flow(self, **kwargs) -> 'ResultProcessor':
             sampler = getattr(optuna.samplers, self._sampler)(seed=self._seed, **kwargs)
         study = optuna.create_study(direction=self._direction, sampler=sampler)
         study.optimize(self._objective, n_trials=self._n_trials)
-        return ResultProcessor(study)
+
+        result_processor = ResultProcessor(study)
+        if self._output_file:
+            result_processor.save_parameters(self._output_file)
+        return result_processor
```

Comment:
I would prefer doing this outside as it was before. The reason is separation of concerns. The optimizer should optimize and the outside world should decide what it wants to do with the results. Putting both in here make code harder to maintain in the long run, since we make the optimizer overly complex.

Reply:
I agree

Reply:
done